src/features/settings/screens/UsersScreen/UsersScreen.tsx

```diff
@@ -16,6 +20,7 @@ type UserDef = {
 
 const UsersScreen = () => {
   const [showEditModal, setShowEditModal] = useState(false);
+  const [showAddModal, setShowAddModal] = useState(false);
```

Comment:
remove add modal state and rename to show user modal
```suggestion
  const [showUserModal, setShowUserModal] = useState(false);
```

Reply:
then you should in the `settings.routes.ts` add a third path for the users screen with a:
```
export enum SettingsPathsEnum {
  ...
  USERS = "/settings/users",
  USERS_EDIT = "/settings/users/:id",
  USERS_CREATE = "/settings/users/create", <-- this is the new one
}

Reply:
and then in the `UsersScreen.tsx` you should extend this useEffect:
```
useEffect(() => {
  setShowUserModal(!!params.id);
}, [params]);
```
to also check if the current path is equal to `SettingsPathsEnum.USERS_CREATE`  and if it is then `setShowUserModal(true)` so it will show the UsersModal